sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -263,6 +267,127 @@ case class GpuStringTrimRight(column: Expression, trimParameters: Option[Express
+      GpuColumnVector.from(ColumnVector.stringConcatenate(sep.getBase, emptyStrScalar,
+        columns.toArray[ColumnView], false), dataType)
+    }
+  }
+
+  private def stringConcatSeparatorVector(columns: ArrayBuffer[ColumnVector],
+      sep: GpuColumnVector): GpuColumnVector = {
+    // GpuOverrides doesn't allow only specifying a separator, you have to specify at
+    // least one value column
+    GpuColumnVector.from(ColumnVector.stringConcatenate(columns.toArray[ColumnView],
+      sep.getBase), dataType)
+  }
+
+  private def resolveColumnVectorAndConcatArrayCols(
+      expr: Expression,
+      numRows: Int,
+      sep: Either[GpuScalar, GpuColumnVector],
+      batch: ColumnarBatch): GpuColumnVector = {
+    withResourceIfAllowed(expr.columnarEval(batch)) {
+      case vector: GpuColumnVector =>
+        vector.dataType() match {
+          case ArrayType(st: StringType, nullable) => concatArrayCol(sep, vector.getBase)
+          case _ => vector.incRefCount()
+        }
+      case s: GpuScalar =>
+        s.dataType match {
+          case ArrayType(st: StringType, nullable) =>
+            // we have to first concatenate any array types
+            withResource(GpuColumnVector.from(s, numRows, s.dataType).getBase) { cv =>
+              concatArrayCol(sep, cv)
+            }
+          case _ => GpuColumnVector.from(s, numRows, s.dataType)
+        }
+      case other =>
+        throw new IllegalArgumentException(s"Cannot resolve a ColumnVector from the value:" +
+          s" $other. Please convert it to a GpuScalar or a GpuColumnVector before returning.")
+      }
+  }
+
+  override def columnarEval(batch: ColumnarBatch): Any = {
+    val numRows = batch.numRows()
+    var sep: Either[GpuScalar, GpuColumnVector] = null
+    try {
+      // get the separator and check for null scalar separator
+      val sepScalarNullOpt = children.head.columnarEval(batch) match {
+        case sepScalar: GpuScalar =>
+          sep = Left(sepScalar)
+          if (sepScalar.getBase.isValid() == false) {
+            // if null scalar separator just return a column of all nulls
+            Some(GpuColumnVector.from(sepScalar, numRows, dataType))
+          } else {
+            None
+          }
+        case sepVec: GpuColumnVector =>
+          sep = Right(sepVec)
+          None
+      }
+      sepScalarNullOpt match {
+        case Some(col) => col
+        case None =>
+          withResource(ArrayBuffer.empty[ColumnVector]) { columns =>
+            children.tail.foreach {
+              columns += resolveColumnVectorAndConcatArrayCols(_, numRows, sep, batch).getBase
+            }
+            sep match {
+              case Left(sepScalar) =>
+                if (columns.size == 1) {
+                  processSingleColScalarSep(columns.head)
+                } else {
+                  stringConcatSeparatorScalar(columns, sepScalar)
+                }
+              case Right(sepVec) => stringConcatSeparatorVector(columns, sepVec)
```

Comment:
FYI a change was just merged into cudf that added an assert for the scalar separator case that broke the java tests. Not sure if there is an assertion for the vector case too (it looks like we don't have unit tests for that)

Reply:
this is handled actually above to process the scalar separator.  If I missed a case let me know